src/routes/posts.tsx

```diff
@@ -1,12 +1,31 @@
-import { PostItem } from "../components/Post/Post.tsx";
+import { PostItem } from "../components/Post/PostItem.tsx";
+import { config } from "../components/Post/post.config.ts";
+import { useGet } from "../hooks/useGet.ts";
+import { useParams } from "react-router-dom";
 import Post from "../components/Post/Post.types.ts";
 
-interface PostList {
-  postsList: Post[];
-}
+export function Posts() {
+  const { postsId } = useParams();
+  const { isLoading, isError, data } = useGet(config.url, postsId);
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (isError) {
+    return <div>Error: Oh no!</div>;
+  }
+  //Changed this to be more meaningful name as "data" is not very descriptive
+  let postsData = data;
+  console.log(postsData);
 
-export function Posts(props: PostList) {
-  return props.postsList.map((post) => (
-    <PostItem key={post.id} text={post.text} id={post.id}></PostItem>
+  return postsData?.map((post: Post) => (
```

Comment:
It's better to declare the type of the response on the hook you're using to access the data, instead of assigning `(post: Post)`. Meaning, TS should already know that `postsData` is `Array<Post>`:

You can pass a generic to `useGet<TData>` and then pass it to `useGet`: (try, I don't remember the exact API)

```
export function useGet<TData>(...) {
   return useQuery<TData>(...)
}
```

Then if you create entity-specific util hooks like I explained above, you'd do:

```
export function usePosts() {
   return useGet<Array<Post>>(...)
}
```

Which then will assign the `Array<Post>` value to `data`, and then the IDE will know that when you do:

```
postsData.map(...)
```

the type of each `post` is `Post`

Reply:
Why not have `useUsers()` with `Array<User>` type + `useUser` (no 's') with `User` type?